truffleruby/src/main/java/org/truffleruby/platform/posix/JNRTrufflePosix.java

```diff
@@ -446,28 +410,77 @@ public int open(CharSequence path, int flags, int perm) {
     @TruffleBoundary
     @Override
     public int write(int fd, byte[] buf, int n) {
+        if ((fd == 1 || fd == 2) && context.getOptions().POLYGLOT_STDIO) {
```

Comment:
I did it this way around as I thought checking the `fd` would be quicker - we aren't in the PE here.